src/java.net.http/share/classes/jdk/internal/net/http/HttpClientImpl.java

```diff
@@ -931,9 +931,17 @@ private void debugCompleted(String tag, long startNanos, HttpRequest req) {
             cf = sendAsync(req, responseHandler, null, null);
             return cf.get();
         } catch (InterruptedException ie) {
-            if (cf != null )
+            if (cf != null) {
+                // get was interrupted, cancels the exchange if possible
                 cf.cancel(true);
-            throw ie;
+                try {
+                    // calling cf.get() again to wait for
+                    // cleanup dependant actions to execute
+                    // this should throw a CancellationException
+                    cf.get();
```

Comment:
Yes, I think you're right. I was thinking about that over the weekend and I don't think calling `get()` here will achieve anything, since the future should already be completed with a cancelletion eception by the time we call get(). If we wanted to wait for the dependent actions to complete what we would need to do is actually cancel the `mexCf` (and not the head of the chain) and wait for the cancellation exception to trickle up. That's not how we implemented MinimalFuture::cancel however, and getting it to do that is more complex than it looks. The way we specified cancel as a best effort allows the cancellation to continue in the backround after returning, so it's actually only an issue for the test where we need to take that into account. I believe what happened with this test failure is that we were on a busy CPU and simply did not wait long enough. So I might just drop the changes to `HttpClientImpl` out of this PR.